azbus/receiver.go

```diff
@@ -294,6 +311,44 @@ func (r *Receiver) receiveMessages() error {
 	}
 }
 
+func (r *Receiver) receiveMessagesInSerial() error {
+
+	r.log.Debugf(
+		"NumberOfReceivedMessages %d, RenewMessageLock: %v",
+		r.numberOfReceivedMessages,
+		r.Cfg.RenewMessageLock,
+	)
+
+	for {
+		err := func() error {
+			var err error
+			var messages []*ReceivedMessage
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			messages, err = r.receiver.ReceiveMessages(ctx, r.numberOfReceivedMessages, nil)
+			if err != nil {
+				azerr := fmt.Errorf("%s: ReceiveMessage failure: %w", r, NewAzbusError(err))
+				r.log.Infof("%s", azerr)
+				return azerr
+			}
+			total := len(messages)
+			r.log.Debugf("total messages %d", total)
+			// we should wrap this in a span ...
+			// XXXX: add spans using something like processMessage/r.handleReceivedMessageWithTracingContext
+			// XXXX: add logic for RenewMessageLock here perhaps
+			err = r.serialHandler.Handle(ctx, messages)
```

Comment:
This seems to take a single *ReceivedMessage but seems like we're passing in []*ReceivedMessage.

Reply:
this is new Handler type defined above afaik, and this one takes a list of messages

Reply:
there are 2 interfaces with Handle() methods with different signatures

Reply:
fixed - only one handler now